### PR TITLE
Cli did support

### PR DESCRIPTION
**Pull request by devin-fisher**
Change the CLI to create DidSigner instead of SimpleSigner. This change required more explicit handling of verkey a few classes.

In the tests there are a large number of changes. Many tests were assuming cryptonyms and changing to DID required a lot of changes.

### CODE REVIEW

File: sovrin_client/agent/walleted.py [hunk 1]
```diff
@@ -923,30 +928,31 @@ def _(reply, err):
...
+                    if PUBKEY in ep:
+                        link.remotePubkey = ep[PUBKEY]
+                    else:
+                        link.remotePubkey = friendlyVerkeyToPubkey(
+                            link.remoteVerkey) if link.remoteVerkey else None
+
+                link.linkLastSynced = datetime.now()
+                self.notifyMsgListener("    Link {} synced".format(link.name))
+            except ValueError:
+                pass
```
Comment (lovesh) [inline sovrin_client/agent/walleted.py]: Too broad of a code block to check for exception. Also need a log message here to help debugging

File: sovrin_client/test/cli/test_nym.py [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
 
 @pytest.fixture("module")
-def trustAnchorSigner():
-    return SimpleSigner()
+def trust_anchor_did_signer():
+    return DidSigner(seed=TRUST_ANCHOR_SEED)
+
+
+@pytest.fixture("module")
+def trust_anchor_cid_signer():
```
Comment (lovesh) [inline sovrin_client/test/cli/test_nym.py]: Why are we testing Cryptonyms if do not intent to use them for identities?

File: sovrin_client/test/cli/test_nym.py [hunk 3]
```diff
@@ -73,14 +61,16 @@ def getNymNotFoundExpectedMsgs(idr):
...
+def testGetDIDWithoutAddingIt(be, do, philCli, trust_anchor_did_signer):
+    ensureConnectedToTestEnv(be, do, philCli)
+    getNym(be, do, philCli, trust_anchor_did_signer.identifier,
+           getNymNotFoundExpectedMsgs(trust_anchor_did_signer.identifier))
 
 
-def testGetCIDWithoutAddingIt(be, do, philCli, trustAnchorSigner):
-    getNym(be, do, philCli, trustAnchorSigner.identifier,
-           getNymNotFoundExpectedMsgs(trustAnchorSigner.identifier))
+def testGetCIDWithoutAddingIt(be, do, philCli, trust_anchor_cid_signer):
```
Comment (lovesh) [inline sovrin_client/test/cli/test_nym.py]: Why test Cryptonyms?

File: sovrin_client/test/cli/test_send_attrib_validation.py [hunk 4]
```diff
@@ -64,18 +64,18 @@ def testSendAttribFailsForNotExistingUuidDest(
        mapper=parameters, expect=ERROR, within=2)
 
 
-def testSendAttribSucceedsForExistingCryptonymDest(
+def testSendAttribSucceedsForExistingDidDest(
```
Comment (lovesh) [inline sovrin_client/test/cli/test_send_attrib_validation.py]: There is not Cryptonym now so rename to `testSendAttribSucceedsForExistingDid`

File: sovrin_client/test/cli/test_nym.py (hunk 3 above)
Comment (devin-fisher) [inline sovrin_client/test/cli/test_nym.py]: Plenum still supports them. So testing them still makes sense. I've changed the CLI to guild you to use DID instead and will only generate key-identifiers using DID but the NYM transaction still supports them so testing that capability still makes sense.

File: sovrin_client/test/cli/test_nym.py (hunk 2 above)
Comment (devin-fisher) [inline sovrin_client/test/cli/test_nym.py]: Plenum still supports them. So testing them still makes sense. I've changed the CLI to guild you to use DID instead and will only generate key-identifiers using DID but the NYM transaction still supports them so testing that capability still makes sense.

**Pull request merged**